contracts/staking/test/unit_tests/mixin_vault_core_test.ts

```diff
@@ -0,0 +1,99 @@
+import { blockchainTests, expect, filterLogsToArguments } from '@0x/contracts-test-utils';
+import { StakingRevertErrors } from '@0x/order-utils';
+import { OwnableRevertErrors } from '@0x/utils';
+
+import { constants } from '../utils/constants';
+
+import {
+    artifacts,
+    TestMixinVaultCoreContract,
+    TestMixinVaultCoreInCatastrophicFailureModeEventArgs,
+    TestMixinVaultCoreStakingContractChangedEventArgs,
+} from '../../src';
+
+blockchainTests.resets('MixinVaultCore', env => {
+    let owner: string;
+    let nonOwnerAddresses: string[];
+    let testContract: TestMixinVaultCoreContract;
+
+    before(async () => {
+        [owner, ...nonOwnerAddresses] = await env.getAccountAddressesAsync();
+
+        testContract = await TestMixinVaultCoreContract.deployFrom0xArtifactAsync(
+            artifacts.TestMixinVaultCore,
+            env.provider,
+            env.txDefaults,
+            artifacts,
+        );
+    });
+
+    describe('Set staking contract', () => {
+        async function testOnlyStakingContractModifierAsync(callerAddress: string): Promise<void> {
+            const tx = testContract.testOnlyStakingContractModifier.callAsync({ from: callerAddress });
+            const expectedError = new StakingRevertErrors.OnlyCallableByStakingContractError(callerAddress);
+            expect(tx).to.revertWith(expectedError);
+        }
+
+        it('Owner can set staking contract', async () => {
+            const newAddress = nonOwnerAddresses[0];
+            const receipt = await testContract.setStakingContract.awaitTransactionSuccessAsync(newAddress, {
+                from: owner,
+            });
+            const eventArgs = filterLogsToArguments<TestMixinVaultCoreStakingContractChangedEventArgs>(
+                receipt.logs,
+                'StakingContractChanged',
+            );
+            expect(eventArgs.length).to.equal(1);
+            expect(eventArgs[0].stakingContractAddress).to.equal(newAddress);
+            expect(await testContract.getStakingContractAddress.callAsync()).to.equal(newAddress);
+            await testContract.testOnlyStakingContractModifier.callAsync({ from: newAddress });
```

Comment:
Nit: I think it would be good to add a comment here to elaborate that this check to verify that the new staking contract can pass the modifier.